lib/src/main/kotlin/com/wire/integrations/jvm/WireAppSdk.kt

```diff
@@ -54,11 +54,19 @@ class WireAppSdk(
             return
         }
         running.set(true)
+
         executor.submit {
             val eventsListener = IsolatedKoinContext.koinApp.koin.get<WireTeamEventsListener>()
             logger.info("Start listening to WebSocket events...")
-            runBlocking(Dispatchers.IO) {
-                eventsListener.connect()
+            while (running.get()) {
+                try {
+                    runBlocking(Dispatchers.IO) {
+                        eventsListener.connect()
+                    }
+                    logger.info("Connection ended, attempting to reconnect...")
+                } catch (e: Exception) {
```

Comment:
```suggestion
                } catch (e: InterruptedException) {
```
In this case you are sure that it will be this subclass of exception.